script.js

```diff
@@ -9,137 +9,343 @@ document.addEventListener("DOMContentLoaded", () => {
+      const tbody = this.createEl({ el: "tbody" });
 
-  const deleteRow = () => {
-    const allCheckboxes = document.querySelectorAll("input");
-    allCheckboxes.forEach((el) => {
-      if (el.checked) {
-        const id = el.getAttribute("id");
-        deleteItem({ url: "https://jsonplaceholder.typicode.com/users", id });
-        el.closest("tr").remove();
+      table.append(caption);
+      table.append(thead);
+      table.append(tbody);
+      thead.append(trHead);
+
+      this.thNames.forEach((title) => {
+        const th = this.createEl({
+          el: "th",
+          text: title,
+          styles: `
+                font-size: 18px;
+                text-transform: uppercase;
+                padding: 10px 0px;
+                `,
+        });
+        trHead.append(th);
+      });
+
+      this.fetchedData.forEach((item) => {
+        const tr = this.createEl({ el: "tr" });
+        this.thNames.forEach((thName) => {
+          const tdDto = {
+            name: thName,
+            body: item.body,
+            cb: this.unique,
+            item,
+          };
+
+          const td = this.createEl({
+            el: "td",
+            text: this.createUniqTd(tdDto),
+            styles: `
+                  padding: 10px;
+                  `,
+          });
+          tr.append(td);
+        });
+        tbody.append(tr);
+      });
+      document.body.prepend(table);
+    }
+
+    unique = (str, isUnique) => {
+      const arr = str.split("").filter((item) => item !== " " && item !== "\n");
+      const objUniq = {};
+      for (let i = 0; i < arr.length; i++) {
+        if (objUniq[arr[i]]) {
+          objUniq[arr[i]] += 1;
+        } else {
+          objUniq[arr[i]] = 1;
+        }
       }
-    });
-  };
+      const uniqArr = Object.values(objUniq);
+      const uniqueLetters = uniqArr.filter((item) => item < 2).length;
+      const moreThanOnce = uniqArr.filter((item) => item > 1).length;
 
-  getData("https://jsonplaceholder.typicode.com/users")
-    .then((data) =>
-      createTable(
-        "Table #1 (Users data)",
-        ["check", "id", "name", "phone", "website"],
-        data
-      )
-    )
-    .catch((err) => {
-      throw new Error(err.message);
+      if (isUnique) {
+        return uniqueLetters;
+      }
+      return moreThanOnce;
+    };
+
+    createUniqTd = (dto) => {
+      const { name, body, cb, item } = dto;
+      switch (name) {
+        case "unique":
+          return cb(body, true);
+        case "more than 1":
+          return cb(body, false);
+        default:
+          return item[name];
+      }
+    };
+  }
+
+  getData("https://jsonplaceholder.typicode.com/posts?_limit=10")
+    .then((data) => {
+      const table = new TableWithUnique({
+        tableCaption: "Table #2 (Posts data)",
+        id: "table2",
+        thNames: ["id", "title", "body", "unique", "more than 1"],
+        fetchedData: data,
+      });
+      return table;
```

Comment:
It isn't a Promise. You don't need to invoke the following method in a separate "then"